### PR TITLE
globus upload workflows

### LINKED ISSUE #2847: When globus option is selcted and the deposit is saved as draft, trigger new job
The new job should:

- See if the user exists in globus via globus client gem

If the user exists in globus:
- Create new endpoint that user will put files into using globus client gem and set permissions correctly
- Email user with this endpoint URL and other instructions (need text from PO)
- Email sdr-support indicating a user has begun a globus deposit with info about the work

If the user does not exist in globus:
- Email user with instructions on how to get started with globus (need text from PO)

### LINKED ISSUE #2855: Add a page/Google doc with instructions for users on how to sign up for Globus and link to it
Users who have never used globus before but want to select that as an upload option need to have logged in at least once so they will be known to the API.  We need to provide them with some instructions that can be linked to from the work edit page if they select that option.

Ask @amyehodge for some verbiage for this page.

### LINKED ISSUE #2871: Handle non-existent Globus user when saving new work as draft
When the user clicks "save as draft" (for the first time only) on a deposit where they selected "use globus", see if the user exists in Globus via `globus_client` gem.

If the user does not exist in globus:

* Redirect to work detail page displaying this message to the user on the top of the screen:
  * > Please check your email for further instructions on how to proceed with your deposit of files into Globus.
* Email user with instructions explaining how to setup Globus and ask them to return to work edit page once done and click "Save As Draft" again when complete.

(If the user does exist, that's covered by a separate ticket: #2847)

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made? 🤔

Fixes #2871 and fixes #2847 and fixes #2855 - handle non-existing and existing globus users saving works as drafts

Adds:

- New job to run all globus setup tasks
- Emails for user needs to activate globus and for sending user globus endpoint link
- State transitions for when user selects globus upload option and back again

To do:
- [x] add actual link to globus documentation google help doc in globus account setup and endpoint emails (from Amy)
- [x] figure how/when/where to send globus endpoint created email
- [x] figure out how to use globus client gem in different environments for development (qa/stage) and troubleshoot
- [x] add globus config/keys to shared configs for all environments
- [x] add test coverage
- [x] integrate globus client gem where needed
- [x] add banner when user needs to activate globus account with button to do this
- [x] add button to dashboard for activating globus account when needed
- [x] show message next to the deposit button when globus selected

Next, we need to do #2884 so that the deposit button is re-enabled for globus uploads when an endpoint exists

NOTE: none of these features are available unless you set the `Settings.globus_upload` feature flag to true (currently set to false).  You can do this in `settings.local.yml`.  Also note that this will attempt to make actual globus API calls, so you will also need to configure the globus credentials in your `settings.local.yml`.  Current values for each environment are in h2 shared configs, e.g. https://github.com/sul-dlss/shared_configs/blob/sul-h2-qa/config/settings/production.yml for QA

## How was this change tested? 🤨

Localhost
Needs tests

### CODE REVIEW

File: app/components/works/buttons_component.html.erb [hunk 1]
```diff
@@ -1,3 +1,11 @@
+<% if Settings.globus_upload %>
+  <div data-deposit-button-target="globusMessage" class="row justify-content-end">
+    <div class="col-8">
+      <p>You can save this deposit as a draft.  Next, you will receive an email with instructions on next steps in Globus.
```
Comment (justinlittman) [inline app/components/works/buttons_component.html.erb:4]: This sentence as "next" in it twice.

File: app/jobs/globus_setup_job.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  def perform(work_version)
+    druid = work_version.work.druid # may be nil
+    user = work_version.work.owner
+    Honeybadger.context({ work_version_id: work_version.id, druid:,
+                          work_id: work_version.work.id, depositor_sunet: user.sunetid })
+
+    if globus_user_exists?(user.email) && work_version.globus_endpoint.blank?
+      # user is known to globus but doesn't have a globus endpoint yet, so create it and send the email
+      create_globus_endpoint(work_version)
+      WorkObserver.globus_endpoint_created(work_version) # send email with globus endpoint
```
Comment (justinlittman) [inline app/jobs/globus_setup_job.rb:17]: Seems odd that this is being invoked through WorkObserver. Typically, WorkObserver is only invoked via the state machine.

File: app/jobs/globus_setup_job.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  def create_globus_endpoint(work_version)
+    user = work_version.work.owner
+    # e.g. 'mjgiarlo/work1234/version1'
+    endpoint_path = "#{user.sunetid}/work#{work_version.work.id}/version#{work_version.version}"
+    success = GlobusClient.mkdir(user_id: user.email, path: endpoint_path)
+
+    # TODO: Error handling?
+    return unless success
```
Comment (justinlittman) [inline app/jobs/globus_setup_job.rb:41]: raise here?

**Review by justinlittman — CHANGES_REQUESTED** [review]
A few minor changes / questions.

File: app/jobs/globus_setup_job.rb (hunk 2 above)
Comment (peetucket) [reply app/jobs/globus_setup_job.rb]: removed `WorkObserver` method and send email directly now from the job

File: app/components/works/buttons_component.html.erb (hunk 1 above)
Comment (peetucket) [reply app/components/works/buttons_component.html.erb]: adjusted text

**Review by justinlittman — APPROVED** [review]

**Pull request merged**